src/orchestration.d.ts

```diff
@@ -0,0 +1,311 @@
+type Payment = unknown;
+type Purse = unknown;
+
+export type TypeUrl = string;
+export type Denom = string; // ibc/... or uist
+
+export type Proto3JSONMsg = {
+  '@type': TypeUrl;
+  value: Record<string, unknown>;
+};
+
+export type EncodeObject = {
+  typeUrl: TypeUrl;
+  value: Uint8Array;
+};
+
+export type ChainAddress = {
+  chainId: string; // 1 for ethereum, cosmoshub-4 for cosmos
+  address: string; // can be bech32 or hex encoded
+};
+
+export type ValidatorAddress = {
+  chainId: string; // 1 for ethereum, cosmoshub-4 for cosmos
+  address: string; // can be bech32 or hex encoded //
+};
+
+/** @throws if not a syntactically valid address */
+export declare function makeValidatorAddress(
+  chainId: string,
+  address: string,
+): ValidatorAddress;
+
+export interface OrchestrationGovernor {
+  registerChain: (chainName: string, connection: string) => Promise<void>;
+}
+
+// chainName: managed like agoricNames. API consumers can make/provide their own
+export interface Orchestrator {
+  getChain: (chainName: string) => Promise<Chain>;
+}
+
+/**
+ * Info for an Ethereum-based chain.
+ */
+export type EthChainInfo = {
+  chainId: string;
+  allegedName: string;
+};
+/**
+ * Info for a Cosmos-based chain.
+ */
+export type CosmosChainInfo = {
+  chainId: string;
+  ibcConnectionInfo: {
+    id: string; // e.g. connection-0
+    client_id: string; // '07-tendermint-0'
+    state: 'OPEN' | 'TRYOPEN' | 'INIT' | 'CLOSED';
+    counterparty: {
+      client_id: string;
+      connection_id: string;
+      prefix: {
+        key_prefix: string;
+      };
+    };
+    versions: { identifier: string; features: string[] }[];
+    delay_period: bigint;
+  };
+  icaEnabled: boolean;
+  icqEnabled: boolean;
+  pfmEnabled: boolean;
+  ibcHooksEnabled: boolean;
+  allowMessages: TypeUrl[];
+  allowQueries: TypeUrl[];
+};
+
+export type ChainInfo = CosmosChainInfo | EthChainInfo;
+
+// marker interface
+interface QueryResult {}
+
+/**
+ * An object for access the core functions of a remote chain.
+ */
+export interface Chain {
+  getChainInfo: () => Promise<ChainInfo>;
+  /**
+   * Provide (get or make) an account on the chain. The account is a
+   * named account associated with the current orchestrator instance
+   * (typically associated with a specific seat). If an account for this `Chain`
+   * with the provided `petName` already exists, it is returned,
+   * otherwise a new account is created on the remote Chain.
+   * @param petName
+   * @returns the account that controls the
+   */
+  provideAccount: (petName?: string) => Promise<OrchestrationAccount>;
+  /* query external chain state */
+  query: (queries: Proto3JSONMsg[]) => Promise<Iterable<QueryResult>>;
+
+  // TODO we need a way to have multiple offers get the same orchestrator.
+}
+
+/**
+ * An object that supports low-level queries and operations for an account on a remote chain.
+ */
+export interface ChainAccount {
+  /**
+   * @returns the address of the account on the chain
```

Comment:
_question_

Account on the local chain or remote chain?

Reply:
The remote / host chain.  There is no concept of an `agoric1...` module account for a ChainAccount on a remote chain. Instead, a `LocalChainAccount` would be used.